Implement conditional logging

chadwhitacre opened the pull request:
Closes #385 by defining a `debug` helper that logs conditionally based on the presence of a `PYBEE_DEBUG` envvar that is set to something other than `'0'`.

- [ ] confirm design
- [ ] test it
- [ ] replace moar commented-out `console.log` calls (also seeing `console.log.info`? is that really a thing?)

chadwhitacre committed:
Start to implement conditional logging

whit537 commented:
Ping @glasnt @freakboy3742 for initial review! :-)

freakboy3742 commented on batavia/batavia.js:
```diff
@@ -6,6 +6,16 @@ BigNumber.config({
 })
 
+// Configure logging.
+if (process.env.PYBEE_DEBUG && process.env.PYBEE_DEBUG !== '0') {
```
BATAVIA_DEBUG would be a better variable name - referring the project, rather than the meta-project.

whit537 replied:
Done in d4d737f60bd16dad4574459cb9f2901cfc5bc762.

freakboy3742 reviewed (changes requested):
Definitely like the idea of abstracting `debug()`. 

Regarding the specific implementation; This is Batavia-specific debug functionality, so I'm wondering if it might be better to put this into the `batavia` namespace (i.e., `batavia.debug()`) to avoid potential collisions. 

There's also the matter of configuration - using process.env is great for deploying inside node, but doesn't help much in the browser. Configuring this as part of [VM initialisation](https://github.com/pybee/batavia/blob/master/batavia/VirtualMachine.js#L15), as something that can be set:

1. by specific argument to the VM instance
2. by inspecting a state variable like `batavia.DEBUG`
3. by inspecting the `process.env` environment

with more specific/local settings taking priority, that would maximise flexibility.

freakboy3742 commented:
Also - the `console.log.info` thing is probably an oversight from some copy-pasta code. I'm a professional. I promise I know what I'm doing. :-)

chadwhitacre committed:
Move to a module

whit537 commented:
> [I]t might be better to put this into the `batavia` namespace (i.e., `batavia.debug()`) to avoid potential collisions.

Attempted in d4d737f60bd16dad4574459cb9f2901cfc5bc762, but I'm getting the following error from `npm run build`. Looks like `batavia/stdlib` is empty. Is there an additional build step I'm missing?

```
ERROR in ./batavia/batavia.js
Module not found: Error: Can't resolve './stdlib' in '/Users/whit537/personal/pybee/batavia/batavia'
 @ ./batavia/batavia.js 24:20-39

ERROR in ./batavia/builtins/__import__.js
Module not found: Error: Can't resolve '../stdlib' in '/Users/whit537/personal/pybee/batavia/batavia/builtins'
 @ ./batavia/builtins/__import__.js 72:25-45
 @ ./batavia/builtins.js
 @ ./batavia/batavia.js
```

Commit:
Fix debug variable not defined  bug

agronauts commented:
@whit537 I am currently working on this, would it be possible for you to give me pushing permissions to your branch or should I make a new pull request?

Commit:
Fix ESLint errors

Commit:
Fix Node version for Travis CI

Commit:
Replace console.log with debug function in comments

whit537 commented:
@agronauts Nice! Invite sent! :-)

Commit:
Correct debug importing

Commit:
Inject debugging variable from django settings.py file

Unfortunately a bit of a hack had to be used to pass server side environment variable in Django
to client side node, however the centralised control it offers is worth it. debug.js was rearranged
so that it accepts some level of debugging.

Commit:
Inject debugging variable from django settings.py file

Unfortunately a bit of a hack had to be used to pass server side environment variable in Django
to client side node, however the centralised control it offers is worth it. debug.js was rearranged
so that it accepts some level of debugging.

Commit:
Merge remote-tracking branch 'whit537/conditional-logging' into conditional-logging

# Conflicts:
#	testserver/testbed.html

Commit:
Merge branch 'master' into conditional-logging

agronauts commented:
@freakboy3742 Could you have another look please? Debugging flag is set in the settings.py file.

agronauts commented:
@freakboy3742 Bump

phildini commented:
Hi there! It looks like this PR might be dead, so we're closing it for now. Feel free to re-open it if you'd like to continue, or think about directing your efforts to https://github.com/beeware/briefcase or https://github.com/beeware/toga. Both of these have more active development right now. 😄
